### ISSUE TITLE
(0,5) Format  "Graphique - Copie d'un champ existant" : valeur arbitraire obligatoire ou colonne existante obligatoire

### ISSUE THREAD

**Issue by sbarreaux2** [Inist-CNRS/lodex#2172]
Lorsqu’on met un graphe en page d’accueil : on le crée dans la page graphe puis en page d’accueil on utilise le format  "Graphique - Copie d'un champ existant". Mais on a alors un triangle orange qui apparait car Lodex attend une valeur arbitraire ou une colonne existante. Donc on met juste un espace pour valeur arbitraire pour éviter que ça apparaisse en erreur !

Est-il possible de trouver une solution élégante qui évite de devoir contourner le système et surtout de chercher des heures d'où vient l'erreur ?

**Comment by yannguillemin** [comment]
Une solution envisageable pourrait être de supprimer "Graphique - Copie d'un champ existant" de la liste et d'ajouter cette fonctionnalité sous la forme d'une icône dans la partie "Valeur source" avec comme demande d'argument, l'identifiant du graphique à cloner.
Reste à voir si cette modification est simple/rapide à faire...

<img width="571" alt="image" src="https://github.com/user-attachments/assets/4b01df19-e9cd-4b32-89ec-fa2d438c1f8b">

**Comment by touv** [comment]
pour ma part, je mets une valeur plus explicite par exemple "voir le détail dans les graphiques" 

à minima, ça pourrait être saisi automatiquement, si la solution proposée par yann n'était pas possible (ou trop coûteuse)

**Comment by yannguillemin** [comment]
Oui, ce serait pas mal d'avoir accès à tous les graphiques (y compris ceux créés en page d'accueil) à partir d'un bouton dédié pour le clonage, cela pourrait ainsi permettre de répondre en même temps à la demande formulée par @AnaelKremer dans la discussion "Possibilité de dupliquer un graphique vers la page d'accueil et inversement" #2091.

**Comment by touv** [comment]
Cette carte pour la solution à minima et une autre carte pour la solution de @yannguillemin

**Comment by Alexmarmelab** [comment]
Solution : mettre une valeur par défaut "Voir le détail dans les graphiques"

**Comment by Alexmarmelab** [comment]
Lorsque je choisis le format "Graphique - Copie d'un champ existant", je n'ai pas de valeur par défaut qui s'ajoute dans "Valeur arbitraire" et quand j'enregistre, j'ai toujours l'erreur sur l'argument requis.

<img width="763" alt="Image" src="https://github.com/user-attachments/assets/02ddbd56-03a7-4a92-a1fe-1004886bbece" />

**Comment by eonm-pro** [comment]
L'erreur persiste toujours après avoir créer un nouveau champ sur la page d'accueil avec un affichage "copie d'un champ existant" :

1. Création d'un nouveau champ : Affichage -> Page d’accueil -> Nouveau champ
2. Configuration du champ : Affichage -> Graphique - Copie d'un champ existant -> Sauvegarder

![Image](https://github.com/user-attachments/assets/1f2d4475-4a0e-4c5d-abf9-341ed7485939)
L'erreur suivante apparait :
![Image](https://github.com/user-attachments/assets/ff8547f3-8bd0-4305-8120-9720f9d8a801)

**Comment by AnaelKremer** [comment]
J'ai créé un graphique, puis dans l'acceuil j'ai voulu le remettre en copie d'un champ existant sans rien renseigner en valeur arbitraire, je n'ai pas de message d'erreur mais le graphique ne s'affiche pas (graph1). En revanche si je rentre un caractère aléatoire en valeur arbitraire le graphique s'affiche cette fois (graph 3). Et si j'entre une valeur arbitraire, enregistre le graphique, puis le modifie en supprimant la valeur arbitraire, j'obtiens une erreur (graph 2).
Dans tous les cas, on n'a plus d'alerte "une transformation est recquise" en back

![Image](https://github.com/user-attachments/assets/beec1e7b-f187-4253-ba30-dded5f2abf4f)

**Comment by jonathanarnault** [comment]
Correction effectuée

**Comment by AnaelKremer** [comment]
Plus d'alerte et graphique bien copié ! OK

**Comment by eonm-pro** [comment]
Testé en 14.0.75 :

Ok pour moi également

**Comment by nantonot** [comment]
V 14.0.75 : ok
